core-bundle/src/Resources/contao/classes/DataContainer.php

```diff
@@ -1583,6 +1583,108 @@ public static function getDriverForTable(string $table): string
 
 		return $dataContainer;
 	}
+
+	/**
+	 * Generates the label for a given data record according to the DCA configuration.
+	 *
+	 * @param array  $row   The data record
+	 * @param string $table The name of the data container
+	 * @param array  $args  The generated labels for each column
+	 */
+	public static function generateRecordLabel(array $row, string $table, array &$args = array()): string
+	{
+		$mode = $GLOBALS['TL_DCA'][$table]['list']['sorting']['mode'] ?? self::MODE_SORTED;
+		$showFields = $GLOBALS['TL_DCA'][$table]['list']['label']['fields'];
```

Comment:
```suggestion
		$showFields = $GLOBALS['TL_DCA'][$table]['list']['label']['fields'] ?? array();
```

Can we be sure that `$GLOBALS['TL_DCA'][$table]['list']['label']['fields']` is always set or should we add a fallback for PHP 8?

Reply:
Currently the DCA assumes that it must be set. See also https://github.com/contao/contao/pull/3644#discussion_r759940288